### PR TITLE
docs: update gateway README

### PR DESCRIPTION
**Pull request by HJianBo**
Fixes <issue-or-jira-number>

### <samp>🤖 Generated by Copilot at 945c6bc</samp>

This pull request enhances the `emqx_gateway` README by improving its structure, content, and style. It aims to provide a better overview of the application's features, architecture, configuration, and usage.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_gateway/README.md [hunk 1]
```diff
@@ -53,6 +79,14 @@ your device protocol and integrate it with EMQX.
 
 Refer to: [ExProto Gateway](../emqx_exproto/README.md)
 
-## Cookbook for emqx_gateway framework
+## Understard emqx_gateway framework
```
Comment (thalesmg) [inline apps/emqx_gateway/README.md:82]:
```suggestion
## Understand emqx_gateway framework
```

File: apps/emqx_gateway/README.md [hunk 2]
```diff
@@ -11,16 +11,42 @@ protocol access on EMQX. For example:
 - Configuration & Schema
 - HTTP/CLI management interfaces
 
-There are some standard implementations available, such as [Stomp](../emqx_stomp/README.md),
-[MQTT-SN](../emqx_mqttsn/README.md), [CoAP](../emqx_coap/README.md),
-and [LwM2M](../emqx_lwm2m/README.md) gateway.
-
 The emqx_gateway application depends on `emqx`, `emqx_authn`, `emqx_ctl` that
 provide the foundation for protocol access.
```
Comment (JimMoen) [inline apps/emqx_gateway/README.md:15]: and also `emqx_authz`
from:
```
apps/emqx_gateway/src/emqx_gateway_ctx.erl
50:    authorize/4
157:-spec authorize(
164:authorize(_Ctx, ClientInfo, PubSub, Topic) ->
165:    emqx_access_control:authorize(ClientInfo, PubSub, Topic).
```

File: apps/emqx_gateway/README.md [hunk 3]
```diff
@@ -11,16 +11,42 @@ protocol access on EMQX. For example:
 - Configuration & Schema
 - HTTP/CLI management interfaces
 
-There are some standard implementations available, such as [Stomp](../emqx_stomp/README.md),
-[MQTT-SN](../emqx_mqttsn/README.md), [CoAP](../emqx_coap/README.md),
-and [LwM2M](../emqx_lwm2m/README.md) gateway.
-
 The emqx_gateway application depends on `emqx`, `emqx_authn`, `emqx_ctl` that
```
Comment (HJianBo) [inline apps/emqx_gateway/README.md:14]:
```suggestion
The emqx_gateway application depends on `emqx`, `emqx_authn`, `emqx_authz`, `emqx_ctl` that
```

**Review by JimMoen — DISMISSED** [review]

**Comment by jinfahua** [comment]
1. link [Implement Gateway via Raw Application](https://github.com/emqx/emqx/blob/9774ad2cdd79abdbb26660a6f4b695a241629b66/apps/emqx_gateway/doc/implement_gateway_via_raw_appliction.md) and [Implement Gateway via Gateway framework](https://github.com/emqx/emqx/blob/9774ad2cdd79abdbb26660a6f4b695a241629b66/apps/emqx_gateway/doc/implement_gateway_via_gateway_framekwork.md) cannot be accessed.
2. The license is not correct, it should be [Apache License](https://github.com/emqx/emqx/blob/master/APL.txt)

File: apps/emqx_gateway/README.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# emqx_gateway
+# Gateway
 
 EMQX Gateway is an application that managing all gateways in EMQX.
```
Comment (sstrigler) [inline apps/emqx_gateway/README.md:3]:
```suggestion
EMQX Gateway is an application framework that manages all gateways within EMQX.
```

File: apps/emqx_gateway/README.md [hunk 5]
```diff
@@ -11,16 +11,42 @@ protocol access on EMQX. For example:
...
+    max_connections = 1024000
+    max_conn_rate = 1000
+  }
+}
+```
 
-## Raw Erlang Application
+## How to develop your Gateway application
+
+There are three ways to develop Gateway application to accept your private protocol
```
Comment (sstrigler) [inline apps/emqx_gateway/README.md:46]:
```suggestion
There are three ways to develop a Gateway application to accept your private protocol
```

File: apps/emqx_gateway/README.md (hunk 4 above)
Comment (HJianBo) [reply apps/emqx_gateway/README.md:3]: Great thanks!!! @sstrigler

**Pull request merged**